File: src/company_brain.py
"""Ephemeral Software Layer — manages the 'Company Brain' files (Core Constraint 2).

The primary assets are context_framework.json and skills.md.
All application code is considered disposable; the brain files are not.
"""
from __future__ import annotations

import json
from pathlib import Path
from typing import Any

COMPANIES_DIR = Path(__file__).parent.parent / "companies"


def company_dir(company_id: str) -> Path:
    d = COMPANIES_DIR / company_id
    d.mkdir(parents=True, exist_ok=True)
    return d


def write_context_framework(company_id: str, context: dict[str, Any]) -> Path:
    path = company_dir(company_id) / "context_framework.json"
    path.write_text(json.dumps(context, indent=2))
    return path


def read_context_framework(company_id: str) -> dict[str, Any]:
    path = company_dir(company_id) / "context_framework.json"
    if not path.exists():
        return {}
    return json.loads(path.read_text())


def write_skills(company_id: str, content: str) -> Path:
    path = company_dir(company_id) / "skills.md"
    path.write_text(content)
    return path


def read_skills(company_id: str) -> str:
    path = company_dir(company_id) / "skills.md"
    if not path.exists():
        return ""
    return path.read_text()


def append_task_log(company_id: str, task_id: str, data: dict[str, Any]) -> None:
    tasks_dir = company_dir(company_id) / "tasks"
    tasks_dir.mkdir(exist_ok=True)
    path = tasks_dir / f"{task_id}.json"
    path.write_text(json.dumps(data, indent=2))

File: src/companyBrain.ts
/**
 * Company Brain — per-venture persistent knowledge.
 * Manages context_framework.json and skills.md for each company instance.
 * These files are the "durable asset" of each venture; source code is ephemeral.
 */
import fs from 'node:fs';
import path from 'node:path';

const COMPANIES_DIR = 'companies';

function ventureDir(companyId: string): string {
  return path.join(COMPANIES_DIR, companyId);
}

function ensureDir(companyId: string): void {
  fs.mkdirSync(ventureDir(companyId), { recursive: true });
}

export function writeContextFramework(companyId: string, context: unknown): void {
  ensureDir(companyId);
  fs.writeFileSync(
    path.join(ventureDir(companyId), 'context_framework.json'),
    JSON.stringify(context, null, 2),
  );
}

export function readContextFramework(companyId: string): Record<string, unknown> {
  const p = path.join(ventureDir(companyId), 'context_framework.json');
  if (!fs.existsSync(p)) return {};
  return JSON.parse(fs.readFileSync(p, 'utf-8')) as Record<string, unknown>;
}

export function writeSkills(companyId: string, content: string): void {
  ensureDir(companyId);
  fs.writeFileSync(path.join(ventureDir(companyId), 'skills.md'), content);
}

export function readSkills(companyId: string): string {
  const p = path.join(ventureDir(companyId), 'skills.md');
  if (!fs.existsSync(p)) return '';
  return fs.readFileSync(p, 'utf-8');
}

export function appendTaskLog(companyId: string, taskId: string, data: unknown): void {
  ensureDir(companyId);
  const p = path.join(ventureDir(companyId), 'task_log.jsonl');
  fs.appendFileSync(p, JSON.stringify({ taskId, ...(data as object) }) + '\n');
}
